ingest/workflow/snakemake_rules/fetch_sequences.smk

```diff
@@ -0,0 +1,135 @@
+"""
+This part of the workflow handles fetching sequences from various sources.
+Uses `config.sources` to determine which sequences to include in final output.
+
+Currently only fetches sequences from GenBank, but other sources can be
+defined in the config. If adding other sources, add a new rule upstream
+of rule `fetch_all_sequences` to create the file `data/{source}.ndjson` or the
+file must exist as a static file in the repo.
+
+Produces final output as
+
+    sequences_ndjson = "data/sequences.ndjson"
+
+"""
+
+
+rule fetch_ncbi_dataset_package:
+    output:
+        dataset_package=temp("data/ncbi_dataset.zip"),
+    retries: 5  # Requires snakemake 7.7.0 or later
+    benchmark:
+        "benchmarks/fetch_ncbi_dataset_package.txt"
+    shell:
+        """
+        datasets download virus genome taxon 12637 \
+            --no-progressbar \
+            --filename {output.dataset_package}
+        """
+
+
+rule extract_ncbi_dataset_sequences:
+    input:
+        dataset_package="data/ncbi_dataset.zip",
+    output:
+        ncbi_dataset_sequences=temp("data/ncbi_dataset_sequences.fasta"),
+    benchmark:
+        "benchmarks/extract_ncbi_dataset_sequences.txt"
+    shell:
+        """
+        unzip -jp {input.dataset_package} \
+            ncbi_dataset/data/genomic.fna > {output.ncbi_dataset_sequences}
+        """
+
+
+def _get_ncbi_dataset_field_mnemonics(wildcards) -> str:
+    """
+    Return list of NCBI Dataset report field mnemonics for fields that we want
+    to parse out of the dataset report. The column names in the output TSV
+    are different from the mnemonics.
+
+    See NCBI Dataset docs for full list of available fields and their column
+    names in the output:
+    https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/command-line/dataformat/tsv/dataformat_tsv_virus-genome/#fields
+    """
+    fields = [
+        "accession",
+        "sourcedb",
+        "isolate-lineage",
+        "geo-region",
+        "geo-location",
+        "isolate-collection-date",
+        "release-date",
+        "update-date",
+        "length",
+        "host-name",
+        "isolate-lineage-source",
+        "bioprojects",
+        "biosample-acc",
+        "sra-accs",
+        "submitter-names",
+        "submitter-affiliation",
+    ]
```

Comment:
There's a lot of duplication of these mappings here - it might be nice to deduplicate